src/foam/nanos/auth/AgentUserAuthService.java

```diff
@@ -63,10 +63,9 @@ public User actAs(X x, User entity) throws AuthenticationException {
       Finds the UserUserJunction object to see if user can act as the passed in user.
       Source (agent) users are permitted to act as target (entity) users, not vice versa.
     */
-    UserUserJunction permissionJunction = (UserUserJunction) agentJunctionDAO_.find(AND(
-      EQ(UserUserJunction.SOURCE_ID, agent.getId()),
-      EQ(UserUserJunction.TARGET_ID, user.getId())
-    ));
+    UserUserJunction permissionJunction = (UserUserJunction) agentJunctionDAO_.find(
+        EQ(UserUserJunction.TARGET_ID, entity.getId())
```

Comment:
Was checking `sourceId` actually redundant? I don't think it was. `agentJunctionDAO` is just an EasyDAO, there's nothing special about it afaict that would be inferring the source id based on the user making the call.

Looking into that also made me realize there's no security on `agentJunctionDAO` at the moment. We've refined `UserUserJunction` to implement the `Authorizable` interface, but we don't actually have an `AuthorizationDAO` decorating the `agentJunctionDAO` to call those methods. So we should add an `AuthorizationDAO` to `agentJunctionDAO`.

All that being said though, I could be wrong or missing something. Did you test this? It seems to me that it'll just return the first junction with a matching targetId, which won't necessarily give you the one you want.

Reply:
You're right. The other relationship DAO's pre-join the DAO, but I see that the Junction DAO doesn't (but probably should). Please revert. sorry

Reply:
Great catch @TW80000, thanks for following up.

Made the change, but can't test on up to date branches till the foam/np conflict is fixed.